python/ray/serve/_private/application_state.py

```diff
@@ -197,26 +197,28 @@ def update(self):
                 finished, pending = ray.wait([self._deploy_obj_ref], timeout=0)
                 if pending:
                     return
+                self._deploy_obj_ref = None
                 try:
                     ray.get(finished[0])
                     logger.info(f"Deploy task for app '{self._name}' ran successfully.")
-                except RayTaskError as e:
+                except Exception as e:
+                    if isinstance(e, RayTaskError):
+                        # NOTE(zcin): we use str(e) instead of traceback.format_exc()
+                        # here because the full details of the error is not displayed
+                        # properly with traceback.format_exc(). RayTaskError has its own
+                        # custom __str__ function.
+                        self._app_msg = f"Deploying app '{self._name}' failed: {str(e)}"
+                    elif isinstance(e, RuntimeEnvSetupError):
+                        self._app_msg = (
+                            f"Runtime env setup for app '{self._name}' failed:\n"
+                            f"{traceback.format_exc()}"
+                        )
+                    else:
+                        self._app_msg = (
+                            "Unexpected error occured while deploying application "
+                            f"'{self._name}':\n{traceback.format_exc()}"
+                        )
```

Comment:
The standard way to do this is:
```suggestion
                except RayTaskError as e:
                        # NOTE(zcin): we use str(e) instead of traceback.format_exc()
                        # here because the full details of the error is not displayed
                        # properly with traceback.format_exc(). RayTaskError has its own
                        # custom __str__ function.
                        self._app_msg = f"Deploying app '{self._name}' failed: {str(e)}"
                except RuntimeEnvSetupError as e:
                        self._app_msg = (
                            f"Runtime env setup for app '{self._name}' failed:\n"
                            f"{traceback.format_exc()}"
                        )
                except Exception as e:
                        self._app_msg = (
                            "Unexpected error occured while deploying application "
                            f"'{self._name}':\n{traceback.format_exc()}"
                        )
```